python/ray/node.py

```diff
@@ -201,7 +201,18 @@ def __init__(
         self._init_temp()
 
         # Validate and initialize the persistent storage API.
-        storage._init_storage(ray_params.storage, is_head=head)
+        storage_uri = ray_params.storage
+        if head:
+            storage._init_storage(storage_uri, is_head=True)
```

Comment:
Should we just prohibit specifying storage if connecting to an existing cluster? This would be consistent with the other Ray params (e.g., num_cpus). It's a bit strange to special case this one.

Reply:
@ericl makes sense. Let me adjust it.

Reply:
@suquark I think this will break the new hot-fix you did for tests.  I think with the new behavior we can simply remove the `storage={}` parts in your hot-fix new tests.

Reply:
yes! thanks for updating these tests!